Filter Ignorable Copyrights #372

adityaviki opened the pull request:
A post-scan plugin to filter out ignorable-copyrights that are read from license's yaml file.

adityaviki commented:
@pombredanne let me know if I need to make any improvements to the code. I'll write the the tests for it afterwards.

sritasngh commented:
Can you please edit the top comment and linked issue no along with your name! This makes easy to understand in that way:)
You could see [here](https://github.com/nexB/scancode-toolkit/pull/2433#issue-591072762) the template. Thank You for your contribution:)

pombredanne commented on docs/source/cli-reference/help-text-options.rst:
```diff
@@ -104,6 +104,8 @@ The Following Help Text is displayed, i.e. This is the help text for Scancode Ve
                                       Files and directories without findings
                                       are omitted (file information is not
                                       treated as findings).
+      --filter-ignoralbe-copyrights   Filter out copyrights that are for the 
```
I think we have a tiny typo:
`--filter-ignoralbe-copyrights` should be `--filter-ignorable-copyrights`
```suggestion
      --filter-ignorable-copyrights   Filter out copyrights that are for the 
```

pombredanne reviewed (changes requested):
Thank you ++
This is looking great! :bow: 
It is missing one thing though before I can merge: some test.
Do you think you can draft one?

Siddhant-K-code committed:
Merge branch 'Siddhant-K-code:siddhant-improving-pypi'

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

adityaviki commented:
> Do you think you can draft one?

Yes of course :)

adityaviki commented:
> Can you please edit the top comment and linked issue no along with your name!

Sorry! I didn't understand. Do I need to add

adityaviki commented:
I didn't mean to close the PR or delete the branch. I wasn't  paying  attention to where I was clicking. :/

pombredanne commented on src/cluecode/plugin_filter_ignorable_copyrights.py:
```diff
@@ -0,0 +1,78 @@
+from commoncode.cliutils import OUTPUT_FILTER_GROUP
```
Could add a header comment matching other files?

pombredanne replied:
```suggestion
#
# Copyright (c) nexB Inc. and others. All rights reserved.
# ScanCode is a trademark of nexB Inc.
# SPDX-License-Identifier: Apache-2.0
# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
# See https://github.com/nexB/scancode-toolkit for support or download.
# See https://aboutcode.org for more information about nexB OSS projects.
#

from commoncode.cliutils import OUTPUT_FILTER_GROUP
```

pombredanne commented:
> I didn't mean to close the PR or delete the branch. I wasn't paying attention to where I was clicking. :/

that's OK :)

adityaviki commented:
@pombredanne I don't know how to solve the failing check.

sritasngh commented:
>I don't know how to solve the failing check.

DCO check is falling because of this https://github.com/nexB/scancode-toolkit/pull/2436/commits/55cea2679297d4f7cd757c8a922b1c0f32906c47 It has co-authour name but signed-off is missing. So what you can do is squash these changes to the original commit of your and add sign-off or you may ask @pombredanne to squash all the commits into one.

adityaviki commented:
@itssingh It's still not clear how to solve the issue.

pombredanne committed:
Add provisional SPDX license ID

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add minimum_coverage to license rules

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new open source license detection rules

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new proprietary license detection rules

Most of these are derived from the MIT and several pose as being MIT
Yet they all have some changes that may be minor in a few cases and
significant in many cases with several proprtietary restrictions.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2695 from nexB/2621-add-ci-docs-aboutfiles

Add CI for Docs and ABOUT files

pombredanne committed:
Merge inishchith:fix_summary_by_facet_for_packages branch

This is for PR #1864 and  #1759

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Fix regex syntax #2686

This was triggering a FutureWarning

Reported-by: Thorsten Godau @tgodau
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge remote-tracking branch 'upstream/develop' into release-preparation

pombredanne committed:
Add new and improve existing license rules

Various GPL 3.0 licenses

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new and improve existing license rules

Various GPL 3.0 licenses

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new bsd license rules

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new bsd license rules

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new proprietary license rules

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new proprietary license rules

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new non-english license rule

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new non-english license rule

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Improve license detection with rules for Apache

Avoid ambiguous license detection as MIT instead of Apache

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Improve license detection with rules for Apache

Avoid ambiguous license detection as MIT instead of Apache

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Correct test expectatations for summarisation

This is for #1759 as a follow up to PR #1864
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2704 from nexB/spdx-2.2

Adopt SPDX v2.2 and fix SPDX TV correctness

pombredanne committed:
Merge remote-tracking branch 'upstream/develop' into release-preparation

pombredanne committed:
Merge pull request #2707 from nexB/copyright-fixes

Improve Copyright detection

pombredanne committed:
Merge remote-tracking branch 'upstream/develop' into release-preparation

pombredanne committed:
Add new license rules

Seen in:
  https://source.chromium.org/chromium/chromium/src/+/main:third_party/fft2d/LICENSE
and inside the code archive seen at:
  http://www.kurims.kyoto-u.ac.jp/~ooura/fft.html

This is in conflict with the web site license which is: takuya-ooura

Reported-by: Henrik Sandklef @hesa
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new license rules

Seen in:
  https://source.chromium.org/chromium/chromium/src/+/main:third_party/fft2d/LICENSE
and inside the code archive seen at:
  http://www.kurims.kyoto-u.ac.jp/~ooura/fft.html

This is in conflict with the web site license which is: takuya-ooura

Reported-by: Henrik Sandklef @hesa
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

agschrei committed:
refactor: turn standalone functions into class methods #1888

Signed-off-by: Alexander Gschrei <alexander.gschrei@fau.de>

agschrei committed:
map multiple authors and use license db url as fallback #1888

Signed-off-by: Alexander Gschrei <alexander.gschrei@fau.de>

pombredanne committed:
Update CHANGELOG with contributors

- Akanksha Garg @akugarg
- Ayan Sinha Mahapatra @AyanSinhaMahapatra
- Jono Yang @JonoYang
- Philippe Ombredanne @pombredanne
- John M. Horan @johnmhoran
- Yunus Rahbar @yns88
- Henrik Sandklef @hesa
- Armijn Hemel @armijnhemel
- Nishchith Shetty @inishchith
- Soim Kim @soimkim
- Bryan Sutula @sutula
- dyh @yunhua-deng
- Mikko Murto @mmurto
- mapelpapel @mapelpapel
- Rainer Bieniek @rbieniek
- gunaztar @gunaztar
- Siddhant Khare @Siddhant-K-code
- Dennis Clark @DennisClark
- Chin-Yeung Li @chinyeungli
- Helio Chissini de Castro @heliocastro

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Support Pypi nameless packages #2465

Improve tests and ensure we can handle nameless PyPI packages.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

agschrei committed:
add external refs for scancode spdx ids #1888

Signed-off-by: Alexander Gschrei <alexander.gschrei@fau.de>

AyanSinhaMahapatra committed:
Modify file level attribute packages to package_manifests

See #2694

Signed-off-by: Ayan Sinha <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Modify tests and test expectations

Modifies tests and test expectations for the change of
file level attribute `packages` to `package_manifests`.

Signed-off-by: Ayan Sinha <ayansmahapatra@gmail.com>

pombredanne committed:
Add versioning approach documentation #2653

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Correct Rst format for misc. doc index

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Use YAMl to display test failures diffs

It is easier to read than JSON in particular for long text lines

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Improve Pypi description cleanup #2465

Description lines are sometimes padded with 8 spaces. When this happen
these need to be trimmed and this was not taking place correctly

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Display outdated version WARNING in scan results #2653

This shows up 90 days after a release date. We now also track the
release date

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Update CHANGELOG

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Move test files committed mistakenly in root dir

In 30b967ab47c96bf these feils were mistakenly pushed in the root dir.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2705 from nexB/release-preparation

Prepare new release

agschrei committed:
docs: break lines at 100 chars length to pass lint check #1888

Signed-off-by: Alexander Gschrei <alexander.gschrei@fau.de>

AyanSinhaMahapatra committed:
Merge branch 'develop' into 2098-top-level-packages

Signed-off-by: Ayan Sinha <ayansmahapatra@gmail.com>

pombredanne committed:
Use tag file to restart release tests

We now create a tag file after each succesful test of a release archive
and this mean we can restart from the last successful tests and not
restest everything. Azure and/or the network are rather flaky lately
so this way we can run the full release matrix even if it means
restarting the test command several times.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Bump release for bug emergency bug fix

Referenced-by: https://github.com/nexB/scancode-toolkit/issues/2713
Reported-by: Thomas Druez <tdruez@nexb.com>
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Require at least commoncode 30.0.0

This is a fix for https://github.com/nexB/scancode-toolkit/issues/2713

Reference: https://github.com/nexB/commoncode/issues/31
Reference: https://github.com/nexB/scancode-toolkit/issues/2713
Reported-by: Thomas Druez <tdruez@nexb.com>
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add tests for many Click versions #2713

Following a commoncode update this adds tests for running scancode
on all supported Click versions.

Reference: https://github.com/nexB/commoncode/issues/31
Reference: https://github.com/nexB/scancode-toolkit/issues/2713
Reported-by: Thomas Druez <tdruez@nexb.com>
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add CHANGELOG

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Use correct path in Click version tests #2713

Reference: https://github.com/nexB/commoncode/issues/31
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2715 from nexB/post-release

Prepare bugfix release 30.0.1 #2713

pombredanne committed:
Return package details in summary #2717

Also do not return empty values in summaries.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

AyanSinhaMahapatra committed:
Add top level `packages` codebase attribute

Adds `packages` codebase attribute and temporarily populate with
all package manifests detected in the codebase.

Signed-off-by: Ayan Sinha <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Modify test expectations for top level `packages`

Signed-off-by: Ayan Sinha <ayansmahapatra@gmail.com>

pombredanne committed:
Bump to 30.1.0

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2718 from nexB/2717-bug-summary

Return package details in summary #2717

AyanSinhaMahapatra committed:
Merge branch 'develop' into 2098-top-level-packages

Signed-off-by: Ayan Sinha <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Restore summarycode and API changes

Restores previously made changes in summarycode, and deprecates
package API with a warning, and adds other small bugfixes.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Fix packages bug for single file resources

Package_manifests were not being collected when there was
a single file in the codebase, this fixes the bug and regenerates tests.

Signed-off-by: Ayan Sinha <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Bump output data version and update CHANGELOG

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

beastrun12j committed:
Update install.rst

pombredanne committed:
Merge pull request #2710 from nexB/2098-top-level-packages

Report `packages` at top level with file level `package_manifests`

pombredanne committed:
Fix character encodings

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2722 from beastrun12j/Updated-install.rst

Update install.rst

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>
Signed-off-by: Pranav Tiwari <pranavtiwari12j@gmail.com>

pombredanne committed:
Remove dangling empty file

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add rare CPOL 1.0 license

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2706 from nexB/omnibus-fall-license-improvements

Omnibus fall license improvements

pombredanne committed:
Merge remote-tracking branch 'upstream/develop' into omnibus-fall-license-improvements

pombredanne committed:
Use correct license expression for SSPL and Elastic

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add first draft ROADMAP

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Refine ROADMAP wording

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Improve license metadata

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

AyanSinhaMahapatra committed:
Rename package attribute metafiles to file_patterns

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Add PackageManifest class to models

Adds a mixin class PackageManifest to models which will have the base
implementations of Package Manifest specific methods for recognizing and
creating ecosystem specific package manifest objects, which will inherit
from this class and the ecosystem Package class.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Create PackageManifest classes for cocoapods

Creates PackageManifest classes for all individual types of cocoapods manifests
and overrides the methods for detection and PackageManifest creation.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Modify package classes to inherit PackageManifest mixin

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Create PackageManifest class fot aboutfiles

Creates PackacgeManifest class for aboutfiles and overrides
detection and PackageMamanifest creation methods.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Modify test expectations for recognize_package_manifests

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Create PackageManifest classes for NPM

Creates PackageManifest classes for NPM manifests
and overrides detection and PackageMamanifest creation methods.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Create PackageManifest classes for phpcomposer

Creates PackageManifest classes for all phpcomposer manifests
and overrides the methods for detection and PackageManifest creation.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Create PackageManifest class for Haxe

Creates PackageManifest classes for haxelib json manifests
and overrides the methods for detection and PackageManifest creation.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Create PackageManifest classes for cargo

Creates PackageManifest classes for rust cargo manifests`
and overrides the methods for detection and PackageManifest creation

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Add PackageManifest class for opam

Creates PackageManifest classes for opam manifests
and overrides the methods for detection and PackageManifest creation.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

mrombout committed:
Add feature to set key phrases in rules

A key phrase can be defined in a rule by surrounding one or more
words (tokens) with the `{{` and `}}` characters. Key phrases are
words that **must** be present in order to successfully match.

The key phrases are parsed for each rule during indexing. Then after
matching using the standard matching algorithm a filter removes all
matches that do not have the key phrases present.

---

At the moment sometimes the `ispan` alone can not be relied upon to determine
if key phrases are present.

At one case stopwords and unknown tokens made it appear as if the `ispan` was
more expansive than it should be. This has been solved by checking if the
`qspan` corresponding with the `ispan` intersect with any unknown tokens or
stopwords.

Another problem that is not solved is with the example boto3 example which
should match with `cc-by-nc-sa-4.0` but it now matching with several other
CC licenses. For some reason `Creative Commons Attribution 4.0 International
License` is considered to be matching in the ispan, and not intersecting with
any unknown tokens or stopwords (citation needed).

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

AyanSinhaMahapatra committed:
Add PackageManifest class for bower

Creates PackageManifest class for bower json manifest
and overrides the methods for detection and PackageManifest creation.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

mrombout committed:
Fix false positive detection of mpl-1.1

License erlangpl-1.1 was detected as mpl-1.1.

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

mrombout committed:
Fix false positive detection of cddl-1.0

License mit was detected as cddl-1.0.

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

mrombout committed:
Fix false positive detection of imagemagick

License apache-2.0 was detected as imagemagick.

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

mrombout committed:
Fix false positive detection of mit

License apache-2.0 was detected as mit.

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

AyanSinhaMahapatra committed:
Create PackageManifest class for freebsd

Creates PackageManifest class for freebsd compact manifest
and overrides the methods for detection and PackageManifest creation.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Add PackageManifest classes for ruby manifests

Creates PackageManifest class for various ruby manifests
and overrides the methods for detection and PackageManifest creation.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Add PackageManifest classes for pypi manifests

Creates PackageManifest class for various python manifests
and overrides the methods for detection and PackageManifest creation.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Create PackageManifest classes for golang manifests

Creates PackageManifest class for go.mod and go.sum manifests
and overrides the methods for detection and PackageManifest creation.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Create PackageManifest class for nuspec manifest

Creates PackageManifest class for nuspec manifest
and overrides the methods for detection and PackageManifest creation.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Add PackageManifest classes for chef manifests

Creates PackageManifest classes for chef manifests
and overrides the methods for detection and PackageManifest creation.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Create PackageManifest classes for conda

Creates PackageManifest classes for conda manifests
and overrides the methods for detection and PackageManifest creation.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Create PackageManifest class for readme manifests

Creates PackageManifest class for readme manifests
and overrides the methods for detection and PackageManifest creation.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Create PackageManifest classes for pubspec

Creates PackageManifest classes for pubspec manifests
and overrides the methods for detection and PackageManifest creation.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Create PackageManifest classes for windows manifests

Creates PackageManifest classes for various windows manifests
and overrides the methods for detection and PackageManifest creation.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Modify PackageManifest classes in build

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Create PackageManifest class for rpm

Creates PackageManifest class for rpm manifests
and overrides the methods for detection and PackageManifest creation.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

petergardfjall committed:
Requalify some notice license rules as references

This changes a number of rules with unqualified license names (such as "Apache 2.0
License") to be classified as (the weaker) "reference" rather than (the
stronger) notice.

In practice, this means that the `is_license_notice` property is changed to
`is_license_reference` for a number of rules.

Signed-off-by: Peter Gardfjäll <peter.gardfjall.work@gmail.com>

dd-jy committed:
Fix false positive where bsd-3-clause is detected as gpl-2.0-plus_and_bootloader-exception

Signed-off-by: Jiyeong Seok <jiyeong.seok@lge.com>

pombredanne committed:
Merge pull request #2737 from nexB/omnibus-fall2-license-improvements

Improve license detection

mrombout committed:
Fix false positive detection of cc-by-4.0

License cc-by-sa-4.0 was detected as cc-by-4.0.

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

mrombout committed:
Fix false positive detection of gpl-2.0

License mit was detected as gpl-2.0.

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

mrombout committed:
Fix false positive of bsd-simplified AND gpl-2.0

Non-license text is detected as bsd-simplified AND gpl-2.0.

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

mrombout committed:
Fix false positive of apache-2.0 AND cc-by-sa-4.0

Non-license text was detected as apache-2.0 AND cc-by-sa-4.0.

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

mrombout committed:
Fix false positive detection of ruby

License mit was detected as ruby.

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

mrombout committed:
Fix false positive detection of apache-2.0

License mpl-2.0 was detected as apache-2.0.

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

mrombout committed:
Fix false positive of apache-2.0 OR epl-2.0

License mit was detected as apache-2.0 OR epl-2.0.

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

mrombout committed:
Fix false positive of apache-2.0 OR gpl-2.0-plus

License Bouncy Castyle License (mit) is detected
as apache-2.0 OR gpl-2.0-plus WITH
classpath-exception-2.0.

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

zacchiro committed:
api.get_licenses: clarify and improve docstring for "min_score" argument

- add a missing "not" to correctly capture min_score semantics
- call the argument "min_score", as per current name
- apply minor style improvements

Signed-off-by: Stefano Zacchiroli <zack@upsilon.cc>

pombredanne committed:
Merge pull request #2763 from zacchiro/feature/get_licenses-docstring

api.get_licenses: clarify and improve docstring for "min_score" argument

pombredanne committed:
Merge branch 'develop' into omnibus-fall2-license-improvements

pombredanne committed:
Do not report template as license #270

These templates as code are used only in the cobra library
The code has not evolved much and is best as a false positive.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new and improved license tests

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new copyleft licenses

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new permissive licenses rules

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new miscellaneous license rules

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new license detection tests

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Promote rule as new license

Used to be a license detection rule

Contributed-by: Dennis Clark <dmclark@nexb.com>
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Promote rule as new license

Used to be a license detection rule

Contributed-by: Dennis Clark <dmclark@nexb.com>
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Improve license metadata

Contributed-by: Dennis Clark <dmclark@nexb.com>
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new licenses

Contributed-by: Dennis Clark <dmclark@nexb.com>
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new license detection tests

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new license tests

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new license detection rules

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Update licenses to SPDX license list 3.15

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Improve license sync

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new proprietary license rule

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

mrombout committed:
Add unit tests to key_phrase_tokenizer

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

mrombout committed:
Replace key_phrase_spans default to use factory

Co-authored-by: Philippe Ombredanne <pombredanne@gmail.com>
Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

pombredanne committed:
Update license metadata

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Fix typo in test data file

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Update SPDX tests with latest list version

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

petergardfjall committed:
revert regen=True change

Signed-off-by: Peter Gardfjäll <peter.gardfjall.work@gmail.com>

pombredanne committed:
Do not use defaultdict for Query unknowns

Do not use defaultdict for Query.unknowns_by_pos and
Query.stopwords_by_pos. Otherwise there are pernicious side effects to
add new entries in these dctionaries when querying them after their
creation.

Reported-by: Mike Rombout @mrombout
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Do not use defaultdict for Query unknowns

Do not use defaultdict for Query.unknowns_by_pos and
Query.stopwords_by_pos. Otherwise there are pernicious side effects to
add new entries in these dctionaries when querying them after their
creation.

Reported-by: Mike Rombout @mrombout
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Correct failing license detection tests

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

mrombout committed:
Explain key phrase syntax in new license rule guide

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

pombredanne committed:
Correct license detection test

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Format functions arguments, black-style

This is much more readable

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

petergardfjall committed:
fix broken tests

Signed-off-by: Peter Gardfjäll <peter.gardfjall.work@gmail.com>

pombredanne committed:
Merge pull request #2759 from petergardfjall/change-notices-to-references

rules with "unqualified" license names are references, not notices

pombredanne committed:
Ensure that all single word rule are references

Some were marked as is_license_tag incorrectly. Also make the case of
the text something which is consistent and likely to be the most common
case style that would show up.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Remove unused arguments and format code

An idx index was used as an agrument in a few places and not used.
Format code and ensure that logging will work

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Check for Rule stored text first in Rule.text()

This helps to update rule text in code and then dump the rule back

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Do not detect some solo word license in binaries

This is a partial implementation of some of the ideas of #2403
Add a new filter_invalid_single_word_matches_in_binaries() function
called in licensedcode.match.refine_matches().

This filters matches under these conditions:

- the match is for a binary file

- the matched rule that has a single word (length 1)

- the matched rule has a low relevance, e.g., under 75

- the matched text has either:
  - one or more leading or trailing punctuations (except for +)
    unless this has a high relevance and the rule is contained as-is
    in the matched text (considering case)

  - mixed upper and lower case charcaters (but not a Title case) unless
    exactly the same mixed case as the rule text

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge remote-tracking branch 'upstream/develop' into omnibus-fall3-license-improvements

pombredanne committed:
Add test for continuous detection #2769

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Rename only_known_words to continuous #2769

This is applying the renaming doone in the code to the actual rules

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

mrombout committed:
Throw InvalidRule when key phrase is not closed

When a key phrase definition is started in a rule (with `{{`) but is
not closed (with `}}`) the rule is considered invalid and an error is
thrown.

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

mrombout committed:
Add key phrases to CHANGELOG

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

mrombout committed:
Add name to AUTHORS.rst

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

AyanSinhaMahapatra committed:
Modify PackageManifest class names and attributes

Modifies PackageManifest class names to be camelcase and deletes
attribute `manifest_type` from individual classes.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Remove duplicate models and update docstrings

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Fix azure pipeline vmimage deprecations

`macos-10.14` and `vs2017-win2016` vm images are deprecated and
this replaces them with newer images.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

mrombout committed:
Update filter_key_phrase_spans to also check qspan

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

mrombout committed:
Add docstring to BasicRule::key_phrase_spans

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

mrombout committed:
Add missing rule for cc-by-nc-sa-4.0

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

mrombout committed:
Remove incompatible key phrase syntax from rules

These rules contained `{{` and `}}` characters, but
were never intended to be considered key phrases.

In order to avoid conflicts I've removed the syntax
from these rules.

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

mrombout committed:
Add @dd-jy to AUTHORS.rst

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

fangxlmr committed:
Enforce to disallow duplicated fields to load licence file

Signed-off-by: Fang Xiaoliang <fangxlmr@foxmail.com>

fangxlmr committed:
Fix invalid license yaml files by resolving duplicated keys

Fixes #2655.

Signed-off-by: Fang Xiaoliang <fangxlmr@foxmail.com>

fangxlmr committed:
Merge duplicated "notes" field value into one

Signed-off-by: Fang Xiaoliang <fangxlmr@foxmail.com>

fangxlmr committed:
Fix invalid rule yaml files by resolving duplicated keys

Signed-off-by: Fang Xiaoliang <fangxlmr@foxmail.com>

pombredanne committed:
Merge pull request #2776 from fangxlmr/fix-yaml-dup-keys

Fix invalid license yaml files by resolving duplicated keys

pombredanne committed:
Merge pull request #2775 from nexB/fix-azure-deprecations

Fix azure pipeline vmimage deprecations

mrombout committed:
Move up key phrase filter to improve results

This will prevent false-positives for the following scenario:

1. There are multiple matches, with a smalller match that id contained
2. It is filtered because of containment in the larger match
3. The larger is discarded because of not having keyphrases
4. There is no match left, we now have a false negative

As a practical example:

1. In query there is `licensed under the AGPL 3.0`
2. In rule1 I have `AGPL 3.0`
3. In rule2 I have `licensed under the {{AGPL 3.0 license}}`
3. The query will initially match both rules
4. The match to rule1 may be discarded as contained in the rule2 match
5. The match to rule2 may be discarded as missing a key phrase

By moving the key phrase filter up before the containment filter the
larger match may be filtered out, giving the smaller matches a chance
to stay.

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

altsalt committed:
fixed missing whitespace in prerequisites list

mrombout committed:
Fix false positive of apache-2.0

Signed-off-by: Mike Rombout <mike.rombout@elastisys.com>

pombredanne committed:
Deprecate license

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

AyanSinhaMahapatra committed:
Remove recognize function from package class

Removes recognize function from Package Class, since
the recognize function is now at PackageManifest class
and this isn't used anywhere.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

pombredanne committed:
Refine license matches filtering. Add new rules

Rename all match filter functions to use more explict names.

Refactor function to set the lines as a LicenseMatch method.

Add misc. new and improved license detection rules.

Improve the order in which some match filters are processed.
For instance this help to ensure that non spurious smaller matches are
not merged and discarded in short spurious matches too early.

Refine non-continuous matches filter for #2769
Rename filter_if_only_known_words_rule() to filter_non_continuous_matches()
Also rename "continuous" Rule field to "is_continuous"

Add new filter_short_matches_scattered_on_too_many_lines() filter
This works by discarding some short matches that are scattered on too
many lines to be a correct match.

Improve overlapping filter for two-token matches that precede or
follow longer matches and overlap only on the word "license". In these
cases, these may be spurious and may be discarded.

Add new and improved license detection rules, and improve existing
license metadata.

Improve code formatting and logging.

Move model fields comments as help text on the model field defeinitions,
such as License and Rule. This will help generate API documentation
later.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Improve French and German copyright detection

Also remove some spurious trailing period.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Format regex code for readability.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Update CHANGELOG

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Do not cap dependent versions

Thank-you-to: Henry Schreiner @henryiii
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Align test expectations with latest code

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge latest develop branch

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Sort and streamline imports

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new build_spdx_license_expression() function

This new function returns an SPDX license expression string from a
ScanCode license_expression string.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2773 from softsense/issue-2637-allow-license-rules-to-require-the-presence-of-certain-defining-keywords

Allow license rules to require the presence of certain defining keywords

pombredanne committed:
Merge pull request #2736 from nexB/roadmap

Add first draft ROADMAP

pombredanne committed:
Refine CycloneDX support #1888

Only deal with an SPDX license_expression and not plain licenses or
declared licenses, because this is the only correct license information.

Removing most type hints since we are not yet checking these.

Remove dependency lookup which could be misleading. Instead add warning
if dependencies are skipped.

Add docstring to most methods and functions.

Streamline nested list comprehensions and filter/map constructions and
replace these with simpler constructs.

Replace JSON encoder by a plain to_dict() serialization throughout.

Move attrs attributes declarations towards the top of classes
declarations rather than at the bottom.

Move move method-like functions to their respective objects, including
moving XML serialization functions to to_xml_element() methods for each
model object.

Use factories for fields defaults such as URN and timestamp.

Split method that work on ScanCode packages lists in two: one operating
on a single package mapping and one working on the packages list.

Update tests to work on larger set of expected data.

Always return all attributes, including empty values for now.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Revert change in SDPX tests XML loading

The ways of XML and encoding are sometimes complex.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge latest develop branch

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new and improved license detection rules

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Refine unknown and stop handling

Rename match.filter_key_phrase_spans to
filter_matches_missing_key_phrases

Move query.Query.tokens_with_unknowns
and query..QueryRun.tokens_with_unknowns
to licensedcode_test_utils

Remove obsolete u'' string prefixes from test_query.py

Add tracing to query.Query.tokens_by_line() processing

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

adityasangave committed:
fix docstring in debain_copyright.py

Signed-off-by: adii21-Ux <adityasangave2112@gmail.com>

pombredanne committed:
Improve LicenseMatch filtering

Call match.set_matched_lines() once centrally

Call key pharse filtering early

Implement function to group matches in regions used for debugging

Implement and track a discard_reason for each LicenseMatch which is a
code that tells why a match is filtered.

Format code

Fix Span and LicenseMatch distance_to for touching spans: it should be
to one then.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

mjherzog committed:
Update ROADMAP.rst

mjherzog committed:
Update ROADMAP.rst

pombredanne committed:
Remove old "rule templates" markers from tests

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Improve tests of LicenseMatch

Improve tests of key phrase testing more and using more than one rule

Add tests for LicenseMatch regions

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Clarify roadmap

Some sentences were left unfinished.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2698 from agschrei/add-cyclonedx-output-option

Add CycloneDx output option

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Improve LicenseMatch fields documentation

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Use license_expression.combine_expressions()

And remove duplicated combine_expressions functions.
Tests have been moved upstream too.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Rename Rule.compute_relevance to set_relevance

Rule.compute_relevance() was only doing a set, hence a renaming to
set_relevance()

Also extract a plain "compute_relevance()" function to use with a length
input only.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Format docstrings

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Use new cache.build_spdx_license_expression()

This returns an SPDX license expression from a plain scancode
expression.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Update CHANGELOG.rst

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge remote-tracking branch 'upstream/develop' into omnibus-fall3-license-improvements

pombredanne committed:
Use license expression strings, not objects

The move to use in the license_expression.combine_expressions()
function missed that it was returning a LicenseExpression when existing
functiosn were returning a string. This is fixing this issue.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Restore combine_expression behavior

This was updated incorrectly when switching to license_expression code
in previous commits.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Remove SPDX license lists false positive rules

We can be smarter and use a filter to catch them all.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

adityasangave committed:
made changes

Signed-off-by: adii21-Ux <adityasangave2112@gmail.com>

pombredanne committed:
Validate and use only lower case license keys

Update rare JTA license
Update rules
Add new Rule and License validation code to ensure that all keys
are lowercase.
Also make minor fixes to docstring and formatting

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Create shared get_licenses_by_spdx_key function

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Improve rule generation

Used shared get_licenses_by_spdx_key() function.
Do not generate false positive

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

soimkim committed:
Remove regular expression futurewarning

Signed-off-by: Soim Kim <soim.kim@lge.com>

pombredanne committed:
Add new filter for false positive license lists

The new filter_false_positive_license_lists_matches() filter is able
to recognize sequences of license matches that are false positive lists
such as a list of SPDX license ids seen in some license related tools.
These were previously detected using a large number of rules that were
subsequences of these lists of licenses.

With this filter we now have removed all the rules used before and
added new rules to properly detect each license reference. As a result,
the license detection is both more accurate (detecting more references)
and less noisy (not reporting false positive lists of sucg references)

- "s" is reinstated as a regular word (and not a stopword)
- several new and improved rules have been added
- tests have been updated accordingly

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new licenses

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Remove unused test options for Python2

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add more license detection rules

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2788 from soimkim/develop

Remove regular expression futurewarning

pombredanne committed:
Merge pull request #2786 from adii21-Ux/docstring_fix

fix docstring in debian_copyright.py

pombredanne committed:
Merge pull request #2778 from altsalt/whitespace-typo-fix

fixes missing whitespace in prerequisites list

Signed-off-by: Wm Salt Hale <salt@altsalt.net>
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2748 from nexB/2098-top-level-packages

Add PackageManifest Class

pombredanne committed:
Add missing test file

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Improve handling of continuous and key phrases

Remove non-continuous filter and use instead the key phrases filter,
as a rule that must be matched continuously is the same as a rule with
a single key phrase covering all its text

Track stopwords on the index side in each rule, and use these to ensure
that when keyphrases are present and contain stopwords we check that the
stopwords are the same on the query and index side of a match to
consider the keyphrases matches.

For stopwords introduced a new index tokenizer
"index_tokenizer_with_stopwords".

Sort stopwords for easier search in the future.

Streamline keyphrase tokenizer and run it in Rule.tokens()

Remove query.Query.stopwords_span that was not used
Remove match.LicenseMatch.qcontains_stopwords() that was not used
Remove match.filter_non_continuous_matches() no longer used

Add new and improved rules, and requalify some non-english short rules
as bona-fide rules.

Apply multiple minor refactoring, doc, and formatting refinements

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Refine license sync

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Refine short Apache license rules and tests

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Refine license tests

And spell spurious with one r, not two.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new draft LicenseDetection

This support new ways to combine multiple license matches together,
and distinguish primary and secondary licenses.
Most concepts have been drawn from early work on Debian copyright
detection improvements.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge remote-tracking branch 'upstream/develop' into improve_license_detection

pombredanne committed:
Merge pull request #2765 from nexB/omnibus-fall3-license-improvements

Add new licenses and new detection rules

pombredanne committed:
Use UNKNOWN_NGRAM_LENGTH constant for ngrams

Rather than using a function argument

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Remove duplicated import

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Format code

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge latest develop branch

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add and update licenses and detection rules

This is a batch of misc. license detection rules and a new license.
These have been mostly found thanks to the upcoming unknown license
detection.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add unknown_licenses argument to get_licenses()

This makes it available in the CLI

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Fix typos in comments and docstring

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Use format for matched license text highlight

This is cleaner and more composable than old-style interpolation

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Enable unknown license match filtering

Add split_weak_matches() function to pre-filter weak unknown matches.
Make unknown matches eligible for filter_spurious_matches() and lower
minimum density to 0.6=5 for longer matches. Move the call for
filter_spurious_matches() earlier in the refine pipeline.

Add new filter_invalid_contained_unknown_matches() function to discard
unknown matches found inside the matched queryregion of larger regular
matches

Extract get_full_qspan_matched_text() function from
get_full_matched_text() for improve reusability. This is designed to be
called when crafting new rules absed on a match (which is what is done
with unknown matches).

Use format for matched license text highlight
This is cleaner and more composable than old-style interpolation.

Improve debug tracing of matched texts.

Apply other minor refactoring and doc impropvements

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Improve UnknownRule

Create unique rule id based on a checksum of the rule content

Also improve key phrases parsing for dnagling {{ {{ braces.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Make match unknown work

Create proper synthethic Rule and LicenseMatch on match and return
a match or None.

Include unknown licenses matching as an option to Index.match

Add tests

Use shorter ngrams of length 6 rather than 7 for better sensitivity
This is balanced by the addition of filters:
- Filter weak unknown matches at match time
- Filter out several weak unknown ngrams at index time

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Improve license rule generation error reporting.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Add new is_generic license flag #1675

Note that this is NOT YET returned in the API and outputs

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge remote-tracking branch 'JRavi2/rename-first-column-in-csv' into develop

pombredanne committed:
Update CHANGELOG

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2592 from akugarg/improve_license_detection

Detect unknown licenses #1675

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Enable regen of HTML test expectations

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

akugarg committed:
Rename copyright keys for consistency #2350

In copyright detection results, rename the "value" attribute to proper
descriptive names, respectively: copyright, holder and author

Signed-off-by: akugarg <akanksha.garg2k@gmail.com>
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Rename copyright keys for consistency #2350

This second commit updates the tests and code that did not exist yet
at the time of the initial patch or code that failed to pass tests.

In copyright detection results, rename the "value" attribute to proper
descriptive names, respectively: copyright, holder and author.

Also:
- rename "copyright_holder" to "holder" in CSV output
- apply minor refactorings

Co-authorred-by: akugarg <akanksha.garg2k@gmail.com>
Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Refactor internal copyright processing

We now use a set of Detection objects to streamline how things are
handled internally.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Merge pull request #2791 from nexB/2350-improve-copyright-handling

Improve copyright handling #2350

pombredanne committed:
Merge  adityaviki:filter_ignorable_copyrights

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne commented:
@adityaviki this will be merged at last in v32.1 ... but it may be partially redundant with pre-existing ScanCode CLI options

pombredanne commented:
https://github.com/nexB/scancode-toolkit/blob/develop/src/cluecode/plugin_filter_clues.py has the feature and predates this PR. Thank you +++, closing.
